src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -1164,8 +1164,13 @@ void MacroAssembler::align(int modulus) {
 }
 
 void MacroAssembler::align(int modulus, int target) {
```

Comment:
How about making both parameters unsigned?
And callers could be changed to something like:
```
align(64, (uint)(uintptr_t)pc() & 63);

```

Reply:
Good idea. It also fixes a couple of conversion warnings. Updated.